index.php

```diff
@@ -20,70 +20,79 @@
-                    
-                    case JSON_ERROR_STATE_MISMATCH:
-                        $strError = 'Underflow or the modes mismatch';
-                        break;
-                    
-                    case JSON_ERROR_CTRL_CHAR:
-                        $strError = 'Unexpected control character found';
-                        break;
-                    
-                    case JSON_ERROR_SYNTAX:
-                        $strError = 'Syntax error, malformed JSON';
-                        break;
-                    
-                    case JSON_ERROR_UTF8:
-                        $strError = 'Malformed UTF-8 characters, possibly incorrectly encoded';
-                        break;
-                    
-                    default:
-                        $strError = 'Unknown error';
-                        break;
-                }
-                
-                $arrRetVal = is_null($config) || !empty($strError) ? [] : $config;
-                break;
-            
-            case 'xml':
-                $config    = simplexml_load_file($strPath);
-                $arrRetVal = empty($config) ? [] : get_object_vars($config);
-                break;
-        }
     }
-    
+
+    switch ($strExtension) {
+
+
+        case 'xml':
+            $config = simplexml_load_file($strPath);
+            $arrRetVal = empty($config) ? [] : get_object_vars($config);
+            break;
+        case 'json':
+        default:
+            $strContents = file_get_contents($strPath);
+            $config = json_decode($strContents, true);
+            $strError = '';
+
+            switch (json_last_error()) {
+                case JSON_ERROR_NONE:
+                    // No code should be put here.
+                    break;
+
+                case JSON_ERROR_DEPTH:
+                    $strError = 'Maximum stack depth exceeded';
+                    break;
+
+                case JSON_ERROR_STATE_MISMATCH:
+                    $strError = 'Underflow or the modes mismatch';
+                    break;
+
+                case JSON_ERROR_CTRL_CHAR:
+                    $strError = 'Unexpected control character found';
+                    break;
+
+                case JSON_ERROR_SYNTAX:
+                    $strError = 'Syntax error, malformed JSON';
+                    break;
+
+                case JSON_ERROR_UTF8:
+                    $strError = 'Malformed UTF-8 characters, possibly incorrectly encoded';
+                    break;
+
+                default:
+                    $strError = 'Unknown error';
+                    break;
+            }
+
+            $arrRetVal = is_null($config) || !empty($strError) ? [] : $config;
+            break;
+    }
+
+
     return $arrRetVal;
 }
 
+ini_set('display_errors', 'on');
+
 // Verify the extensions are loaded before running.
 if (!extension_loaded('pgsql')) {
     echo "Postgresql not enabled: you need the 'pgsql' module.\n";
-    exit(1);
+    //exit(1);
```

Comment:
The <code>pgsql</code> extension is required, since it runs the actual <code>COPY</code> command.
Side note: the <code>pgsql</code> extension is not deprecated.